src/bulk/common.ts

```diff
@@ -923,6 +923,32 @@ export interface BulkWriteOptions extends CommandOperationOptions {
   forceServerObjectId?: boolean;
 }
 
+/**
+ * TODO(NODE-XXX)
+ * BulkWrites merge complexity is implemented in executeCommands
+ * This provides a vehicle to treat bulkOperations like any other operation (hence "shim")
+ * We would like this logic to simply live inside the BulkWriteOperation class
+ * @internal
+ */
+class BulkWriteShimOperation extends AbstractOperation {
+  bulkOperation: BulkOperationBase;
+  constructor(options: BulkWriteOptions, bulkOperation: BulkOperationBase) {
```

Comment:
usually "bucket" options are the last parameter passed to a constructor, any particular reason to have this flipped here?

Reply:
No reason, I'll flip!